### PR TITLE
Update Rspec test

### PR DESCRIPTION
**Pull request by emmanuelsteven**
in this PR,
- Did unit test for rentals.rb
-  persons.rb
- teacher.rb
- base_decorator.rb
- students.rb
- classroom.rb
- nameable.rb
- capitalize_decorator.rb etc
PLEASE NOTE I AM WORKING SOLO
Screenshot 2023-07-19

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  context 'When you create a book' do
+    let(:book) { Book.new('Legend of the seeker', 'Openhiemer') }
+
+    it 'should return correct title' do
+      expect(book.title).to eq('Legend of the seeker')
+    end
+
+    it 'should return correct author' do
+      expect(book.author).to eq('Openhiemer')
+    end
```
Comment (danmainah) [inline spec/book_spec.rb:14]: - [ ] @emmanuelsteven Great work adding tests for the book class, kindly consider adding tests for the to_hash and add_rental methods inside the book class.

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    let(:person) { Person.new(19, 'ijola', parent_permission: true) }
+
+    it 'should return the correct name' do
+      expect(person.correct_name).to eq('ijola')
+    end
+
+    it 'should determine if the person can use services' do
+      expect(person.can_use_services).to eq(true)
+    end
+  end
```
Comment (danmainah) [inline spec/person_spec.rb:39]: - [ ]  @emmanuelsteven Great work adding tests for the person class, kindly consider adding tests for the to_hash and add_rental methods inside the person class.

File: spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    end
+  end
+
+  context 'When calling methods on a student' do
+    let(:student) { Student.new([], 24, 'jasper', parent_permission: true) }
+
+    it 'should return hooky symbol' do
+      expect(student.play_hooky).to eq('¯\(ツ)/¯')
+    end
+  end
```
Comment (danmainah) [inline spec/student_spec.rb:41]: - [ ]  @emmanuelsteven Great work adding tests for the student class, kindly consider adding tests for the to_hash and classroom methods inside the student class.

File: spec/teacher_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    end
+
+    it 'should return correct name' do
+      expect(teacher.name).to eq('jasper')
+    end
+
+    it 'should return correct condition' do
+      expect(teacher.can_use_services?).to be(true)
+    end
+  end
```
Comment (danmainah) [inline spec/teacher_spec.rb]: - [ ]  @emmanuelsteven Great work adding tests for the teacher class, kindly consider adding tests for the to_hash  method inside the student class.

File: spec/rental_spec.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      expect(title).to eq('Legend')
+      expect(date).to eq('2023')
+    end
+
+    it 'should show that "rentals" in Book and Person are updated' do
+      expect(@person.rentals.length.positive?).to eq(true)
+      expect(@book.rentals.length.positive?).to eq(true)
+      expect(@person.rentals[0].person.name).to eq('Sam')
+    end
+  end
```
Comment (danmainah) [inline spec/rental_spec.rb]: - [ ]  @emmanuelsteven Great work adding tests for the rental class, kindly consider adding tests for the to_hash  method inside the student class.

Comment (danmainah) [reply spec/rental_spec.rb]: - [ ]  @emmanuelsteven Great work adding tests for your classes, however you for tot add tests for that you created while implementing the Decorator design pattern, this is part of the project requirements as shown below
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @emmanuelsteven , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added tests for all classes required in project instruction except the decorating classes.
✔️  Ensured all tests added are passing.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Kindly consider adding more social links in your GitHub authors section, this will make it easier for users to reach out to you.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by DeliceLydia — APPROVED** [review]
## Hi @emmanuelsteven, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**